olp-cpp-sdk-core/src/cache/DiskCache.cpp

```diff
@@ -236,7 +246,13 @@ OpenResult DiskCache::Open(const std::string& data_path,
   }
 
   database_.reset(db);
-  return OpenResult::Success;
+
+  if (!CheckDbState()) {
```

Comment:
This should be only done in r/o mode. The mutable cache is opened and closes in all kinds of states and compaction is working there as it has a r/w env.